Program.cs

```diff
@@ -0,0 +1,113 @@
+﻿// See https://aka.ms/new-console-template for more information
+
+using System;
+using System.Runtime.Intrinsics.X86;
+
+public class Stack <T>
+{
+    public T[] stack;
+    private int top; // top points to the current element
+    public int capacity; 
+    //public int Capacity;
+
+    public Stack(int capacity)
+    {
+        this.capacity = capacity;
+        stack = new T[capacity];
+        top = -1;
+    }
+
+    private bool isEmpty()
+    {
+        if (top == -1)
+        {
+            return true;
+        }
+        else return false;
+    }
+
+    private bool isFull()
+    {
+        if (top == capacity - 1) return true;
+        else return false;
+    }
+
+    public void Push (T t) // Stores an object on top of the stack.
+    {
+        if (!isFull())
+        {
+            top = top + 1;
+            stack[top] = t;
+        }
+        else return;
+        
+    } 
+    public T Pop() //  Removes the object on top of the stack and returns it.
+    {
+        if (!isEmpty())
+        {
+            T removed = stack[top];
+            top = top - 1;
+            capacity = capacity - 1;
+            return removed;
+        }
+        else
+        {
+            return default(T);
+        }
+    }
+    public void Print () // Prints out all stack objects.
+    {
+        // this method prints the stack starting from the top element (last element)
+
+        int temp = top;
```

Comment:
temp variable naming is very generic.
wouldn't a reverse for loop work here also?